src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs

```diff
@@ -785,7 +785,25 @@ private bool IsValidExtensionMethodSignature()
             }
 
             var parameter = parameters[0];
-            return (parameter.RefKind == RefKind.None) && !parameter.IsParams;
+            if (parameter.IsParams)
+            {
+                return false;
+            }
+            else if (parameter.RefKind == RefKind.Ref)
```

Comment:
> else if (parameter.RefKind == RefKind.Ref) [](start = 11, length = 43)

I am not sure this is the right place for checks like these because we still can run into a method that violates the conditions and is defined in source. If we want to remove the methods from consideration, we probably should do this during lookup or during overload resolution.